pkg/providers/amifamily/al2.go

```diff
@@ -34,14 +34,34 @@ type AL2 struct {
 }
 
 // SSMAlias returns the AMI Alias to query SSM
-func (a AL2) SSMAlias(version string, instanceType *cloudprovider.InstanceType) string {
-	amiSuffix := ""
-	if !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceNVIDIAGPU]) || !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceAWSNeuron]) {
-		amiSuffix = "-gpu"
-	} else if instanceType.Requirements.Get(v1.LabelArchStable).Has(v1alpha5.ArchitectureArm64) {
-		amiSuffix = fmt.Sprintf("-%s", v1alpha5.ArchitectureArm64)
+func (a AL2) DefaultAMIs(version string) []SSMAliasOutput {
+	return []SSMAliasOutput{
+		// AMD64
+		{
+			Name:  "amazon-linux-2",
+			Query: fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended/image_id", version),
+			Requirements: scheduling.NewRequirements(
```

Comment:
With this set of requirements, a GPU instance would be compatible with this, which isn't what you want. You need a `NotIn` requirement to exclude any nodes that have nvidia or neuron accelerators

Reply:
This needs to be properly validated with testing too, since you are changing logic on the common AMI provider codepath ^

Reply:
That would be covered by the tests that are in the e2etests to make sure the right AMIs are picked. right?

Reply:
Can you double-check that we have tests for getting the right AMI for GPU/non-GPU workloads?